security/basic/src/test/java/io/quarkus/ts/openshift/security/basic/BasicSecurityIT.java

```diff
@@ -77,6 +78,86 @@ public void admin(String username, String password, boolean known) {
                 .body(ok ? equalTo("Hello, admin " + username) : anything());
     }
 
+    @Test
+    public void customPolicyOnAnnotations() {
+        given()
+                .auth().basic("alice", "rabbit")
+                .get("/custom/annotated")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("Hello, user alice"));
+
+        given()
+                .auth().basic("bob", "builder")
+                .get("/custom/annotated")
+                .then()
+                .statusCode(HttpStatus.SC_FORBIDDEN);
+        given()
+                .auth().basic(USER_USERNAME, USER_PASSWORD)
+                .get("/custom/annotated")
+                .then()
+                .statusCode(HttpStatus.SC_FORBIDDEN);
+        given()
+                .auth().basic(ADMIN_USERNAME, ADMIN_PASSWORD)
+                .get("/custom/annotated")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("Hello, user albert"));
+    }
+
+    @Test
+    public void customPolicyProperties() {
+        given()
+                .auth().basic("alice", "rabbit")
+                .get("/custom/properties")
+                .then()
+                .statusCode(HttpStatus.SC_FORBIDDEN);
+
+        given()
+                .auth().basic("bob", "builder")
+                .get("/custom/properties")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("Hello, user bob"));
+        given()
+                .auth().basic(USER_USERNAME, USER_PASSWORD)
+                .get("/custom/properties")
+                .then()
+                .statusCode(HttpStatus.SC_FORBIDDEN);
+        given()
+                .auth().basic(ADMIN_USERNAME, ADMIN_PASSWORD)
+                .get("/custom/properties")
+                .then()
+                .statusCode(HttpStatus.SC_OK)
+                .body(equalTo("Hello, user albert"));
+    }
+
+    @Test
```

Comment:
Could you add here some info on the policy precedence over the other to clarify it?

Reply:
Ok,  got it, thanks